tst/data/proxy/UserProfileStoreTest.java

```diff
@@ -1,5 +1,66 @@
 package data.proxy;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import data.structure.UserProfile;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.junit.Test;
+
 public class UserProfileStoreTest {
-    // TODO
+
+    /**
+     * Tests null return for missing UserProfile id.
+     */
+    @Test
+    public void testGetProfileNonExistent() {
+        TestUserProfileStore userProfileStore = new TestUserProfileStore();
+        Map<String, UserProfile> userProfiles = new HashMap<String, UserProfile>();
+        userProfileStore.setUserProfiles(userProfiles);
```

Comment:
I wasn't aware you could access private vars in a JUnit test class :+1:  - addressed.